### DISCUSSION TITLE
getaddrinfo ENOENT error in services with the custom plugin lib

### DISCUSSION THREAD

**Discussion by Noste92** [mia-platform/community#264] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v11.1.2

### Description

Hi,

we are investigating an issue in our projects with different microservices written in node with the Custom Plugin Lib and the Flow Manager.

Sometimes the http calls come to an error due to a name resolution error.

For example here's an exception when the Flow Manager tries to persist the saga calling the crud-service:
```json
{
	"level": 50,
	"time": 1686120341763,
	"pid": 7,
	"hostname": "flow-manager-service-856c5cc598-jb45d",
	"errorStack": "Error: getaddrinfo ENOENT crud-service\n    at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:109:26)\n    at GetAddrInfoReqWrap.callbackTrampoline (node:internal/async_hooks:130:17)",
	"sagaId": "ID_SAGA_1",
	"msg": "unable to upsert saga having id: ID_SAGA_1"
}
```
from the logs it seems to be an error with name resolution, but it's strange that:
- it happens sporadically, without a clear reason.
- if we try to resolve the host `crud-service` from the pod with the command `nslookup crud-service` works just fine.

We are a little bit stuck with the analisys of this problem, have you got any suggestion about it?

**Comment by fredmaggiowski** [comment]
does the crud service have any restart when this occurs? This seems like a problem related with the infrustructure more than an problem with the node applications 🤔

**Reply by Noste92** [reply]
Hi @fredmaggiowski ,

we didn't see any restarts when it occurs, after the investigation we have realized this:
- this problems occurs on many pods, also when we try to reach an host external to the cluster
- we don't think it's related to the DNS, in this case we should have a EAI_AGAIN error, and if we call an invalid host we should have a ENOTFOUND error

I agree with you that it could be related with the infrastructure, I have opened the discussion just to know if someone encountered this issue in the past or have some tips about it.

**Reply by Noste92** [reply, marked as answer]
Hi @fredmaggiowski,

After further investigations we have identified the cause of the problem in Dynatrace.
By disabling the service the problem disapears.
We have asked for support to the team who manage Dynatrace.

Thank you for your reply, I'll mark this message as answer because this problem it's not related to Mia Platform.

**Reply by fredmaggiowski** [reply]
Great to know it, thanks for the info
